apps/thanksgivingday/thanksgiving_day.star

```diff
@@ -16,29 +16,30 @@ R0lGODlhIAAgAPcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 """)
 
 def main(config):
-    timezone = config.get("timezone") or "America/New_York"
-    now = time.now().in_location(timezone)
-
-    thanksgiving_year = now.year
-
-    thanksgiving_day = math.ceil(28 - (5 + now.year + now.year / 4 - now.year / 100 + now.year / 400) % 7)
-    if thanksgiving_year == 2024:
-        thanksgiving_day = 28
-    elif thanksgiving_year == 2025:
-        thanksgiving_day = 27
+    def getthanksgivingday(year):
+        nov30 = time.time(year = year, month = 11, day = 30, location = timezone)
+        day_of_week = humanize.day_of_week(nov30)
+        calc = day_of_week - 4
+        if calc >= 0:
+            day = 30 - calc
+        else:
+            day = 30 - (calc + 7)
 
-    if 11 == now.month:
-        if 0 > thanksgiving_day - now.day:
-            thanksgiving_year = now.year + 1
+        return day
 
-    if now.month > 11:
-        thanksgiving_year = now.year + 1
-
-    thanksgiving = time.time(year = thanksgiving_year, month = 11, day = thanksgiving_day, hour = 0, minute = 0, location = timezone)
-
-    print(thanksgiving)
-    days_til_thanksgiving = math.ceil(time.parse_duration(thanksgiving - now).seconds / 86400)
-    print(days_til_thanksgiving)
+    timezone = config.get("timezone") or "America/New_York"
```

Comment:
Should this be `timezone = config.get("$tz", "America/New_York")`?

Reply:
I’m gonna be honest, I have no idea. Looks to be that way ($tx) in some of the other countdown apps (ie Christmas) but it’s this way (timezone) in the Halloween countdown. Not sure which way is the correct way.

Reply:
The Halloween countdown app was fixed in https://github.com/tidbyt/community/pull/2854.

Reply:
Ah i see, I'll go ahead and update it then